src/lib/core/overlay/scroll/scroll-dispatcher.ts

```diff
@@ -19,26 +20,28 @@ export class ScrollDispatcher {
   /** Subject for notifying that a registered scrollable reference element has been scrolled. */
   _scrolled: Subject<void> = new Subject<void>();
 
+  /** Keeps track of the global `scroll` and `resize` subscriptions. */
+  _globalSubscription: Subscription = null;
+
+  /** Keeps track of the amount of subscriptions to `scrolled`. Used for cleaning up afterwards. */
+  private _scrolledCount = 0;
+
   /**
    * Map of all the scrollable references that are registered with the service and their
    * scroll event subscriptions.
    */
   scrollableReferences: Map<Scrollable, Subscription> = new Map();
 
-  constructor() {
-    // By default, notify a scroll event when the document is scrolled or the window is resized.
-    Observable.fromEvent(window.document, 'scroll').subscribe(() => this._notify());
-    Observable.fromEvent(window, 'resize').subscribe(() => this._notify());
-  }
-
   /**
    * Registers a Scrollable with the service and listens for its scrolled events. When the
    * scrollable is scrolled, the service emits the event in its scrolled observable.
    * @param scrollable Scrollable instance to be registered.
    */
   register(scrollable: Scrollable): void {
     const scrollSubscription = scrollable.elementScrolled().subscribe(() => this._notify());
+
     this.scrollableReferences.set(scrollable, scrollSubscription);
+    this._addGlobalSubscription();
```

Comment:
Should we still be adding/removing the global subscription in register and deregister? I thought that we would only worry about them through the `scrolled` function